OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -181,6 +182,31 @@ void Frame::attachGeometry(OpenSim::Geometry* geom)
     updProperty_attached_geometry().adoptAndAppendValue(geom);
 }
 
+void Frame::scale(const SimTK::Vec3& scaleFactors)
+{
+    for (int i = 0; i < getProperty_attached_geometry().size(); ++i) {
+        Geometry& geo = upd_attached_geometry(i);
+        geo.upd_scale_factors() = geo.get_scale_factors()
+                                  .elementwiseMultiply(scaleFactors);
+    }
+}
+
+void Frame::scale(const SimTK::State& s, const ScaleSet& scaleSet)
+{
+    Super::scale(s, scaleSet);
+
+    if (getProperty_attached_geometry().size() == 0)
+        return;
+
+    // Get scale factors for base frame (if an entry for the base frame exists).
+    const int idx = scaleSet.getIndexBySegmentName(findBaseFrame().getName());
+    if (idx < 0)
+        return;
+    const Vec3& scaleFactors = scaleSet[idx].getScaleFactors();
+
+    scale(scaleFactors);
```

Comment:
Consider renaming this function to `scaleAttachedGeometry()`. As-is, I'm confusing it with the more general "scale()" function.

Reply:
Done. Also, this 1-argument `Frame::scale()` method would be overloading the virtual `scale()` on ModelComponent, which smells like the same problem I encountered with `Model::scale()`.